homeassistant/components/elmax/switch.py

```diff
@@ -19,27 +19,27 @@ class ElmaxSwitch(ElmaxEntity, SwitchEntity):
     @property
     def is_on(self) -> bool:
         """Return True if entity is on."""
-        if self.transitory_state is not None:
-            return self.transitory_state
-        return self._device.opened
+        if self._transitory_state is not None:
+            return self._transitory_state
+        return self.elmax_coordinator.get_actuator_state(
+            self._device.endpoint_id
+        ).opened
 
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the entity on."""
-        client = self._coordinator.http_client
-        await client.execute_command(
+        await self.elmax_client.execute_command(
             endpoint_id=self._device.endpoint_id, command=SwitchCommand.TURN_ON
         )
-        self.transitory_state = True
-        await self.async_update_ha_state()
+        self._transitory_state = True
+        self.async_write_ha_state()
```

Comment:
We should not need this, instead call the coordinator for an update

Reply:
I see that your point, however that is still necessary. I should have stated this in a comment within the code: apologies if I made it unclear. Let me explain why this is necessary.

The reason why I added a protected `_transitory_state` variable is to deal with the current implementation of Elmax Cloud HTTP Api.
When an actuation command is issued against the cloud, it is handled asynchronously by the cloud as it immediately returns "OK, taken into account". The actuator might then take up to 2-3 seconds to issue the command and align the Elmax Cloud state accordingly. This means that fetching the state from HTTP API (i.e. calling a coordinator update) just after a command has been sent causes a "glitch": the HTTP API will still serve the "old" state (which will soon change).

In an attempt to mitigate such issue, I've decided to introduce a new "_transitory_state" variable. This variable is set when a command is issued locally and will take over the coordinator state. When the coordinator fetches a new state, it will blank the _transitory_state so that entities won't take that into account and will instead rely on the coordinator data.

Ok, true, this still does not make sure there aren't race conditions (after all there's little we can do as this reaquires changes on the Elmax API) but surely limits the glitching state.

Reply:
> When an actuation command is issued against the cloud, it is handled asynchronously by the cloud as it immediately returns "OK, taken into account". The actuator might then take up to 2-3 seconds to issue the command and align the Elmax Cloud state accordingly. This means that fetching the state from HTTP API (i.e. calling a coordinator update) just after a command has been sent causes a "glitch": the HTTP API will still serve the "old" state (which will soon change).

This is not fixed by the current workaround created, as you'll never know when the coordinator is polling. For example, it could be the second after you have changed it, thus still miss it.

Reply:
As I stated, that's a mitigation attempt which won't solve the glitch but mitigataes it. You suggestion was to trigger a state update immediately after the command has been issued: that definetely triggers the race and produces the glitch. Using this transitory state will reduce the posibility this happens, not prevent it from happening.

The only neat way to solve this issue is to have a sort of "transaction" ticket from the cloud call, which can be later checked out to see if that succeeded or not. Or, even better, have a synchronous behaviour from the Elmax cloud, which we cannot pretend at this time, I guess...

Any other idea to better handle this?

Reply:
yes, see my comment above :)

Reply:
> Instead, you should maybe wait for a state change to be confirmed? I know the Verisure has such a construct, but not sure if that fits here

I've given a look at the Verisure integration. As I thought, it has something Elmax does not: "transaction ids". When you issue an ARM command, the cloud returns a transaction ID which can be used to later track the outcome of the command. The current Elmax API _does not provide that_. 

![image](https://user-images.githubusercontent.com/4648843/145294283-72f95734-d910-43f0-8124-abc034f3aed6.png)

I cannot apply the same logic without am ID identifying the exact command we have issued. 
For a moment, let's assume we try to handle this without a transaction ID but just saving the previous state and assume that will change sooner or later due to our command. Here's what would happen in the best case scenario:

0. Actuator is OFF
1. HA issues the TURN_ON COMMAND to the Elmax Cloud: under the hood we save the previous state somewhere. The UI correctly puts the switch into ON state (as the toggle entity does).
2. HA polls the cloud a first time and it returns OFF (old state), but thanks to the new state check mechanism, we ignore that (as we expect a state change to take place)
3. we poll the cloud a second time, and it now returns ON (new state): that makes it work.

Now, what happens if another user (via the web-interface, the Elmax APP or phisical access button on the panel) triggers another command (OFF) between 2 and 3? It will cause a stall which can be only "recovered" with some sort of  gives up timeout which assumes a double command was triggered. Is this acceptable (and preferrable) to you?

Reply:
Following your advise, I've implemented something close to the approach implemented by Verisure. 
The new implementation triggers the command and starts waiting until the state changes at coordinator level. Of course I had to set a timeout to give up in case of stalls (for instance, due to concurrent updates and double commands).

This allows me to get rid of the _transitory_state and simplifies even more the integration itself.

Please let me know what you think about this new approach.